src/back/auth/src/service/tokens_session.cpp

```diff
@@ -23,23 +32,36 @@ struct jwt_session_impl {
 	jwt::algorithm::rs256 _algo;
 };
 
-Session::Session(const std::string& privateKeyPath)
+Session::Session(
+		const components::ComponentContext& ctx,
+		const std::string& privateKeyPath)
 {
-	const auto key = readKey(privateKeyPath);
-	jwt::algorithm::rs256 algo{"", key, "", ""};
-
-	auto verifier = jwt::verify()
-		.allow_algorithm(algo)
-		.with_issuer(std::string{_issuer})
-		.leeway(60UL); // value in seconds, add some to compensate timeout
+	changeKey(privateKeyPath);
+  	auto& fs_task_processor = ctx.GetTaskProcessor("fs-task-processor");
+
+	_task = utils::Async(fs_task_processor, "key_watching_task", [privateKeyPath, this] {
+			while (!engine::current_task::ShouldCancel()) {
+				auto inotify = new engine::io::sys_linux::Inotify();
+				inotify->AddWatch(privateKeyPath, engine::io::sys_linux::EventType::kModify);
+				auto event = inotify->Poll(engine::Deadline());
+				if (event) {
+					LOG_WARNING() << "Session key changed";
+					changeKey(privateKeyPath);
+				}
+			}
+        }
+    );
+}
 
-	_jwt = std::make_shared<jwt_session_impl>(std::move(verifier), std::move(algo));
+Session::~Session() {
+	_task.SyncCancel();
 }
 
 std::string Session::Create(
 		const std::string& userId,
 		const std::string& sessionId)
 {
+	std::shared_lock<engine::SharedMutex> lock(_mutex);
```

Comment:
Тип указывать не обязательно если ты передаёшь значение.
```suggestion
	std::shared_lock lock(_mutex);
```